### PR TITLE
Fixing MongoDB backend to work properly with TTL

### PR DESCRIPTION
**Pull request by mpuhacz**
This PR fixes an issue with MonogDB backend which caused ignoring TTL indexes. Interestingly, the previous implementation used a keyword argument for pymongo collection update method that hasn't existed. Besides that, we are also bumping pymongo version to the latest one.

Some of appveyor tests should fail: #1035

### CODE REVIEW

**Comment by mpuhacz** [comment]
Some of the tests seem to be failing due to network problems during `before_install` phase. Could you please rerun CI?

**Comment by thedrow** [comment]
I don't know anyone who is using the MongoDB broker. I don't mind merging this though.
In any case, the minimum version should be what the MongoDB [result backend](https://github.com/celery/celery/blob/master/requirements/extras/mongodb.txt) supports.

File: .travis.yml [hunk 1]
```diff
@@ -9,16 +9,19 @@ env:
...
   - test
   - lint
-  
+
 
 matrix:
   include:
     - python: 2.7
       env: TOXENV=2.7
+      before_install: sudo apt-get update && sudo apt-get install libgnutls-dev
```
Comment (thedrow) [inline .travis.yml]: Why do we need to re-add this?

File: requirements/extras/mongodb.txt [hunk 2]
```diff
@@ -1 +1 @@
-pymongo>=2.6.2,<3.0
+pymongo>=3.8.0
```
Comment (thedrow) [inline requirements/extras/mongodb.txt]: Version should be `>=3.3.0`.

File: .travis.yml (hunk 1 above)
Comment (mpuhacz) [reply .travis.yml]: Travis build was failing without this package. See: https://travis-ci.org/celery/kombu/jobs/568362312

**Comment by mpuhacz** [comment]
@thedrow Thanks for the review. MongoDB backend is quite useful especially if you don't want to integrate additional components into your backend. As long as the message throughput is kept on a reasonable level Mongo is doing pretty well.

**Pull request merged**